# BME-Term-Project

**Code descriptions (libraries, output, and running)**

Summary: 
FOXO3 sequences are extracted from Uniprot, Ensembl, AND NCBI, in that order.
If a sequence isn't found in Uniprot it's then searched in Ensembl, and if not found there
it searches NCBI last. All NCBI sequences are inputted in BLAST to check for orthology using human FOXO3 sequence.

The sequences are then split into 3 files, the full file has the entire sequences, the DBD file
has only the DNA binding domain, using the coordinates given by the human FOXO3 sequence in uniprot (157-251).
Finally the IDR file has all of the other (likely disordered regions).

The MSA file takes the sequence files from Data_extraction.py, aligns them using MAFFT (Kazutaka and Daron 2013). Then alignments are trimmed using TrimAl (in order to reduce noise) (Capella-Gutiérrez et al 2009). The test run will output every file (for manually checking quality), so there will be the alignment files before trimming, and the full final sequencs after alignment and trimming. The full run will only output final sequences.

MAFFT alignment parameters
- DBD sequence: mafft --localpair --genafpair --maxiterate 1000 --reorder --adjustdirection --thread N infile *L-INS-i
- IDR sequence and full sequence:  --genafpair --maxiterate 1000 --ep 0 --adjustdirection --thread N infile *E-INS-i

I chose *L-INS-i for the DBD sequences because it's likely the most accurate, and since the DBD region is highly conserved I was certain alignment would be pretty clean (Kazutaka and Daron 2013). For the whole sequence, and IDR sequence I decided to use *E-INS-i with --ep 0 because there are a lot of disordered regions which might contribute to excess gaps (Kazutaka and Daron 2013). Both of these are accuracy oriented, which I prioritized since I only have a sample size of 73 species (Kazutaka and Daron 2013). 

I utilized HMMer just to verify orthology, and if alignments are clean. I ran them for both the DBD and full sequence, keeping in mind that the disordered regions in the full sequence might affect results. I only saved the file for my DBD HMMer and posted it below. From this file I observed that EFFN was very low (2.55) for a sample size of 72, this reveals that the sequences are all very similar. This confirms (I think) that my sequences are very conserved, orthologous and likely correctly aligned. My full HMM_build file output, had an EFFN = 0.681702, for 73 sequences and was probably affected by IDR regions.

I replicated the code created by Osbourne and others in python to create my phylogenetic tree. I use a distance based method called neighbor joining. I felt this was a good choice considering again, my smaller 73 sequence sample size.

The res.py code from the paper on p53 seems applicable to my case, so I used it to create a csv file of residue specific scores.

Then I will get a file of normalized lifespan. Lifespan, RES file, and Phylogenetic tree will be used with a PGLS code for a final analysis. (hopefully theres something).
**


# Dataset_extraction.py

FOXO3 sequences are extracted from Uniprot, Ensembl, AND NCBI, in that order.
If a sequence isn't found in Uniprot it's then searched in Ensembl, and if not found there
it searches NCBI last. All NCBI sequences are inputted in BLAST to check for orthology using human FOXO3 sequence.

The sequences are then split into 3 files, the full file has the entire sequences, the DBD file
has only the DNA binding domain, using the coordinates given by the human FOXO3 sequence in uniprot (157-251).
Finally the IDR file has all of the other (likely disordered regions).



*Libraries:*

- requests
- time
- urllib.parse
- subprocess
- tempfil
- Bio (biopython) 
- sys

*Test run (only 5 species):*

```
python3 Dataset_extraction.py test
```

*Full run (all species):*

```
python3 Dataset_extraction.py full
```

*Output ({name} = full or test):*

DISPLAYING TEST EXAMPLES ONLY

- {name}_foxo3_seq.fasta
  
```
>Homo_sapiens
MAEAPASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
EDDEDDEDGGGRAGSAMAIGGGGGSGTLGSGLLLEDSARVLAPGGQDPGSGPATAAGGLS
GGTQALLQPQQPLPPPQPGAAGGSGQPRKCSSRRNAWGNLSYADLITRAIESSPDKRLTL
SQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHNLSLHSRFMRVQNEGTGKSSWWIINPDG
GKSGKAPRRRAVSMDNSNKYTKSRGRAAKKKAALQTAPESADDSPSQLSKWPGSPTSRSS
DELDAWTDFRSRTNSNASTVSGRLSPIMASTELDEVQDDDAPLSPMLYSSSASLSPSVSK
PCTVELPRLTDMAGTMNLNDGLTENLMDDLLDNITLPPSQPSPTGGLMQRSSSFPYTTKG
SGLGSPTSSFNSTVFGPSSLNSLRQSPMQTIQENKPATFSSMSHYGNQTLQDLLTSDSLS
HSDVMMTQSDPLMSQASTAVSAQNSRRNVMLRNDPMMSFAAQPNQGSLVNQNLLHHQHQT
QGALGGSRALSNSVSNMGLSESSSLGSAKHQQQSPVSQSMQTLSDSLSGSSLYSTSANLP
VMGHEKFPSDLDLDMFNGSLECDMESIIRSELMDADGLDFNFDSLISTQNVVGLNVGNFT
GAKQASSQSWVPG

>Mus_musculus
MAEAPASPVPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEEDDDEDDEDGG
GRASSAMVIGGGVSSTLGSGLLLEDSAMLLAPGGQDLGSGPASAAGALSGGTPTQLQPQQPLPQPQPGAA
GGSGQPRKCSSRRNAWGNLSYADLITRAIESSPDKRLTLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIR
HNLSLHSRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRRAVSMDNSNKYTKSRGRAAKKKAALQAAPESA
DDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTVSGRLSPILASTELDDVQDDDGPLSPMLYSSS
ASLSPSVSKPCTVELPRLTDMAGTMNLNDGLAENLMDDLLDNIALPPSQPSPPGGLMQRGSSFPYTAKSS
GLGSPTGSFNSTVFGPSSLNSLRQSPMQTIQENRPATFSSVSHYGNQTLQDLLASDSLSHSDVMMTQSDP
LMSQASTAVSAQNARRNVMLRNDPMMSFAAQPTQGSLVNQNLLHHQHQTQGALGGSRALSNSVSNMGLSD
SSSLGSAKHQQQSPASQSMQTLSDSLSGSSLYSASANLPVMGHDKFPSDLDLDMFNGSLECDMESIIRSE
LMDADGLDFNFDSLISTQNVVGLNVGNFTGAKQASSQSWVPG



>Gallus_gallus
MKNSIRHNLSLHSRFIRVQNEGTGKSSWWMINPDGGKVGKAPRRRAVSMDNSNKYTKSRG
RAAKKKAALQTAQEASEDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTISGRLS
PILASTELDDVQDDDAPLSPMLYSSPSSLSPSVNKPCTVELPRLTDMAGTMNLNDGLTDN
LMDDLLDNITLPSSQQSPTGGMMQRSSSFPYGSKGSGLGSPSSSFNNAVFGPSSLNSLRQ
SPMQTIQENKQATFSSISHYNNQTLQDLLASDALSHSDVMMTQSDPLMSQASTAVSAQNS
RRNIMLRNDPMMSFAAQSSQGGLVNQSLPHHQHQSHSSPLSGSRALSNSISNIGLNDSNS
LGSKHQQSPVNQSMQTLSDPLSGSSLYSSSMNLPVMGHEKFPSDLDLDIFNGSLECDMES
IIRSELMDADGLDFNFDSLISAQNVVSLNVGNFTGAKQASSQSWVPG


>Danio_rerio
MAETTLEPLSSLDIAIDPDFEPQKRPRSCTWPLPESSMAKPASNDTDIIPEEEDDEDDSA
MTINANGMGEGEDNGSPSLAEELISINGQENTGSPLSSQAASATTASPEVASQQQTPRKS
SSRRNAWGNLSYADLITKAIESTPDKRLTLSQIYDWMVSSVPYFKDKGDSNSSAGWKNSI
RHNLSLHSRFIRVQNEGTGKSSWWMINPEGGRGGKAPRRRAVSMDNGNKYTKTARGRAAK
KKAALQAAQEGSLENISAGGLSKWPGSPTSRSSDEQESCWTDFRSRTNSNASTVSGRLSP
ILANPELDDVSDGPPLSPILYSSPSSLSPPSSTSSNSKTCPGELPVLADLTGTMNLNEAD
NLMVDLLDNIATQVQVVQESSPFSFGSKSTASPPSTSTSQTVSNNTGGFSNPIFGPSPGS
LRQKPMQTIQENKQTSFSSISCFETSLQDLLNSDSLVSHSHSDVMMTQSDPLISQASAAV
TSQNYRLRNNPMLRNDPMMSSFSSVLNGRSGHLQNKQHHQGSNGQGSLRSLSNNRAQSLV
NDANNLVSAKQLLQPQPTMLMTETPLYSGMNGSNGLGICHPSGTDRFPSDLDLDIFNGSL
DCDVDSIIRSELMDSDGLDFNFDALMQNAVSLNPVGNFTGTKQSNHSWVPG

>Anolis_carolinensis
MCILSFLSQNSIRHNLSLHSRFIRVQNEGTGKSSWWVINPDGGKGGKAPRRRAVSMDNSN
KYTKSRGRAAKKKASLQAAQETTEDSPTQLSKWPGSPTSRSSDELEAWTDFRSRTNSNAS
TISGRLSPILASTELDEVQDDDAPLSPMLYSSPPSMSPSVNKPCNVELPRLTDMAGTMNL
NDGLTDNLMDDLLDNISLPPSQQSPTGGLMQRSSSFPYSSKGSGLGSPSSTFNNGVFGPS
SLNSLRQSPMQTIQENKQATFSSISHYNNQTLQDLLASDSHSHSDVMMTQSDPLMSQAST
AVAAQNVRRSIMLRNDPMMSFAAQPSQGSLVNQNLLHHQHQSQNSSLSGSRALSNSVGTM
GLSDSNSIGPAKHQQQSPANHSMQSLSDSLSGSSLYSNSLSLPVMGHDKFPSDLDLDIFN
GSLECDMESIIRSELMDADGLDFNFDSLISAQNVVTLNVGNFTGAKQASSQSWVPG
```
  
- {name}_foxo3_DBD.fasta

```
>Homo_sapiens_DBD
WGNLSYADLITRAIESSPDKRLTLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHNLSL
HSRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRRA
>Mus_musculus_DBD
GNLSYADLITRAIESSPDKRLTLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHNLSLH
SRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRRAV
>Gallus_gallus_DBD
CTVELPRLTDMAGTMNLNDGLTDNLMDDLLDNITLPSSQQSPTGGMMQRSSSFPYGSKGS
GLGSPSSSFNNAVFGPSSLNSLRQSPMQTIQENKQ
>Danio_rerio_DBD
MVSSVPYFKDKGDSNSSAGWKNSIRHNLSLHSRFIRVQNEGTGKSSWWMINPEGGRGGKA
PRRRAVSMDNGNKYTKTARGRAAKKKAALQAAQEG
>Anolis_carolinensis_DBD
SPSVNKPCNVELPRLTDMAGTMNLNDGLTDNLMDDLLDNISLPPSQQSPTGGLMQRSSSF
PYSSKGSGLGSPSSTFNNGVFGPSSLNSLRQSPMQ
```

- {name}_foxo3_IDR.fasta 

```
>Homo_sapiens_IDR
MAEAPASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
EDDEDDEDGGGRAGSAMAIGGGGGSGTLGSGLLLEDSARVLAPGGQDPGSGPATAAGGLS
GGTQALLQPQQPLPPPQPGAAGGSGQPRKCSSRRNAVSMDNSNKYTKSRGRAAKKKAALQ
TAPESADDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTVSGRLSPIMASTELDE
VQDDDAPLSPMLYSSSASLSPSVSKPCTVELPRLTDMAGTMNLNDGLTENLMDDLLDNIT
LPPSQPSPTGGLMQRSSSFPYTTKGSGLGSPTSSFNSTVFGPSSLNSLRQSPMQTIQENK
PATFSSMSHYGNQTLQDLLTSDSLSHSDVMMTQSDPLMSQASTAVSAQNSRRNVMLRNDP
MMSFAAQPNQGSLVNQNLLHHQHQTQGALGGSRALSNSVSNMGLSESSSLGSAKHQQQSP
VSQSMQTLSDSLSGSSLYSTSANLPVMGHEKFPSDLDLDMFNGSLECDMESIIRSELMDA
DGLDFNFDSLISTQNVVGLNVGNFTGAKQASSQSWVPG
>Mus_musculus_IDR
MAEAPASPVPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
DDDEDDEDGGGRASSAMVIGGGVSSTLGSGLLLEDSAMLLAPGGQDLGSGPASAAGALSG
GTPTQLQPQQPLPQPQPGAAGGSGQPRKCSSRRNAWSMDNSNKYTKSRGRAAKKKAALQA
APESADDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTVSGRLSPILASTELDDV
QDDDGPLSPMLYSSSASLSPSVSKPCTVELPRLTDMAGTMNLNDGLAENLMDDLLDNIAL
PPSQPSPPGGLMQRGSSFPYTAKSSGLGSPTGSFNSTVFGPSSLNSLRQSPMQTIQENRP
ATFSSVSHYGNQTLQDLLASDSLSHSDVMMTQSDPLMSQASTAVSAQNARRNVMLRNDPM
MSFAAQPTQGSLVNQNLLHHQHQTQGALGGSRALSNSVSNMGLSDSSSLGSAKHQQQSPA
SQSMQTLSDSLSGSSLYSASANLPVMGHDKFPSDLDLDMFNGSLECDMESIIRSELMDAD
GLDFNFDSLISTQNVVGLNVGNFTGAKQASSQSWVPG
>Gallus_gallus_IDR
MKNSIRHNLSLHSRFIRVQNEGTGKSSWWMINPDGGKVGKAPRRRAVSMDNSNKYTKSRG
RAAKKKAALQTAQEASEDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTISGRLS
PILASTELDDVQDDDAPLSPMLYSSPSSLSPSVNKPATFSSISHYNNQTLQDLLASDALS
HSDVMMTQSDPLMSQASTAVSAQNSRRNIMLRNDPMMSFAAQSSQGGLVNQSLPHHQHQS
HSSPLSGSRALSNSISNIGLNDSNSLGSKHQQSPVNQSMQTLSDPLSGSSLYSSSMNLPV
MGHEKFPSDLDLDIFNGSLECDMESIIRSELMDADGLDFNFDSLISAQNVVSLNVGNFTG
AKQASSQSWVPG
>Danio_rerio_IDR
MAETTLEPLSSLDIAIDPDFEPQKRPRSCTWPLPESSMAKPASNDTDIIPEEEDDEDDSA
MTINANGMGEGEDNGSPSLAEELISINGQENTGSPLSSQAASATTASPEVASQQQTPRKS
SSRRNAWGNLSYADLITKAIESTPDKRLTLSQIYDWSLENISAGGLSKWPGSPTSRSSDE
QESCWTDFRSRTNSNASTVSGRLSPILANPELDDVSDGPPLSPILYSSPSSLSPPSSTSS
NSKTCPGELPVLADLTGTMNLNEADNLMVDLLDNIATQVQVVQESSPFSFGSKSTASPPS
TSTSQTVSNNTGGFSNPIFGPSPGSLRQKPMQTIQENKQTSFSSISCFETSLQDLLNSDS
LVSHSHSDVMMTQSDPLISQASAAVTSQNYRLRNNPMLRNDPMMSSFSSVLNGRSGHLQN
KQHHQGSNGQGSLRSLSNNRAQSLVNDANNLVSAKQLLQPQPTMLMTETPLYSGMNGSNG
LGICHPSGTDRFPSDLDLDIFNGSLDCDVDSIIRSELMDSDGLDFNFDALMQNAVSLNPV
GNFTGTKQSNHSWVPG
>Anolis_carolinensis_IDR
MCILSFLSQNSIRHNLSLHSRFIRVQNEGTGKSSWWVINPDGGKGGKAPRRRAVSMDNSN
KYTKSRGRAAKKKASLQAAQETTEDSPTQLSKWPGSPTSRSSDELEAWTDFRSRTNSNAS
TISGRLSPILASTELDEVQDDDAPLSPMLYSSPPSMTIQENKQATFSSISHYNNQTLQDL
LASDSHSHSDVMMTQSDPLMSQASTAVAAQNVRRSIMLRNDPMMSFAAQPSQGSLVNQNL
LHHQHQSQNSSLSGSRALSNSVGTMGLSDSNSIGPAKHQQQSPANHSMQSLSDSLSGSSL
YSNSLSLPVMGHDKFPSDLDLDIFNGSLECDMESIIRSELMDADGLDFNFDSLISAQNVV
TLNVGNFTGAKQASSQSWVPG
```

# MSA.py

The MSA file takes the sequence files from Data_extraction.py, aligns them using MAFFT (Kazutaka and Daron 2013). Then alignments are trimmed using TrimAl (in order to reduce noise) (Capella-Gutiérrez et al 2009). The test run will output every file (for manually checking quality), so there will be the alignment files before trimming, and the full final sequencs after alignment and trimming. The full run will only output final sequences.

MAFFT alignment parameters
- DBD sequence: mafft --localpair --genafpair --maxiterate 1000 --reorder --adjustdirection --thread N infile *L-INS-i
- IDR sequence and full sequence:  --genafpair --maxiterate 1000 --ep 0 --adjustdirection --thread N infile *E-INS-i

I chose *L-INS-i for the DBD sequences because it's likely the most accurate, and since the DBD region is highly conserved I was certain alignment would be pretty clean (Kazutaka and Daron 2013). For the whole sequence, and IDR sequence I decided to use *E-INS-i with --ep 0 because there are a lot of disordered regions which might contribute to excess gaps (Kazutaka and Daron 2013). Both of these are accuracy oriented, which I prioritized since I only have a sample size of 73 species (Kazutaka and Daron 2013). 

*Libraries*

- os
- shutil
- subprocess

*Test run (only 5 species):*

```
python3 MSA.py test
```

*Full run (all species):*

```
python3 MSA.py full
```

*TEST RUN Output files:*
- test_full_seq_aligned.fasta - ONLY ALIGNED
  
```
>Homo_sapiens
MAEAPASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
EDDEDDEDGGGRAGSAMAIG--GGGGSGTLGSGLLLEDSARVLAPGGQDPGSGPATAAGG
LSGGTQALLQPQQPLPPPQPGAAGGSGQPRKCSSRRNAWGNLSYADLITRAIESSPDKRL
TLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHNLSLHSRFMRVQNEGTGKSSWWIINP
DGGKSGKAPRRRAVSMDNSNKYTKS-RGRAAKKKAALQTAPESADD--SPSQLSKWPGSP
TSRSSDELDA-WTDFRSRTNSNASTVSGRLSPIMASTELDEVQDDDAPLSPMLYSSSASL
SP-----SVSKPCTVELPRLTDMAGTMNLNDGLTENLMDDLLDNITLPPSQPSPTGGLMQ
RSSSFPYTTKGSG----------LGSPTSSFNSTVFGPSSLNSLRQSPMQTIQENKPATF
SSMSHYGNQTLQDLLTSDSL---SHSDVMMTQSDPLMSQASTAVSAQN--SRRNVMLRND
PMM-SFAAQPNQGSLVNQNLLHHQHQTQ-GALGGSRALSNSVSNMGLSESSSLGSAKHQQ
QSPVSQSMQTLSDSLSGSSLYSTSANL-PVMGHEKFPSDLDLDMFNGSLECDMESIIRSE
LMDADGLDFNFDSLISTQNVVGLN-VGNFTGAKQASSQSWVPG
>Mus_musculus
MAEAPASPVPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
DDDEDDEDGGGRASSAMVI---GGGVSSTLGSGLLLEDSAMLLAPGGQDLGSGPASAAGA
LSGGTPTQLQPQQPLPQPQPGAAGGSGQPRKCSSRRNAWGNLSYADLITRAIESSPDKRL
TLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHNLSLHSRFMRVQNEGTGKSSWWIINP
DGGKSGKAPRRRAVSMDNSNKYTKS-RGRAAKKKAALQAAPESADD--SPSQLSKWPGSP
TSRSSDELDA-WTDFRSRTNSNASTVSGRLSPILASTELDDVQDDDGPLSPMLYSSSASL
SP-----SVSKPCTVELPRLTDMAGTMNLNDGLAENLMDDLLDNIALPPSQPSPPGGLMQ
RGSSFPYTAKSSG----------LGSPTGSFNSTVFGPSSLNSLRQSPMQTIQENRPATF
SSVSHYGNQTLQDLLASDSL---SHSDVMMTQSDPLMSQASTAVSAQN--ARRNVMLRND
PMM-SFAAQPTQGSLVNQNLLHHQHQTQ-GALGGSRALSNSVSNMGLSDSSSLGSAKHQQ
QSPASQSMQTLSDSLSGSSLYSASANL-PVMGHDKFPSDLDLDMFNGSLECDMESIIRSE
LMDADGLDFNFDSLISTQNVVGLN-VGNFTGAKQASSQSWVPG
>Gallus_gallus
M-----------------------------------------------------------
------------------------------------------------------------
------------------------------------------------------------
----------------------------KNSIRHNLSLHSRFIRVQNEGTGKSSWWMINP
DGGKVGKAPRRRAVSMDNSNKYTKS-RGRAAKKKAALQTAQEASED--SPSQLSKWPGSP
TSRSSDELDA-WTDFRSRTNSNASTISGRLSPILASTELDDVQDDDAPLSPMLYSSPSSL
SP-----SVNKPCTVELPRLTDMAGTMNLNDGLTDNLMDDLLDNITLPSSQQSPTGGMMQ
RSSSFPYGSKGSG----------LGSPSSSFNNAVFGPSSLNSLRQSPMQTIQENKQATF
SSISHYNNQTLQDLLASDAL---SHSDVMMTQSDPLMSQASTAVSAQN--SRRNIMLRND
PMM-SFAAQSSQGGLVNQSLPHHQHQSHSSPLSGSRALSNSISNIGLNDSNSLGS-KH-Q
QSPVNQSMQTLSDPLSGSSLYSSSMNL-PVMGHEKFPSDLDLDIFNGSLECDMESIIRSE
LMDADGLDFNFDSLISAQNVVSLN-VGNFTGAKQASSQSWVPG
>Anolis_carolinensis
M-----------------------------------------------------------
------------------------------------------------------------
------------------------------------------------------------
-----------CILSFLS----------QNSIRHNLSLHSRFIRVQNEGTGKSSWWVINP
DGGKGGKAPRRRAVSMDNSNKYTKS-RGRAAKKKASLQAAQETTED--SPTQLSKWPGSP
TSRSSDELEA-WTDFRSRTNSNASTISGRLSPILASTELDEVQDDDAPLSPMLYSSPPSM
SP-----SVNKPCNVELPRLTDMAGTMNLNDGLTDNLMDDLLDNISLPPSQQSPTGGLMQ
RSSSFPYSSKGSG----------LGSPSSTFNNGVFGPSSLNSLRQSPMQTIQENKQATF
SSISHYNNQTLQDLLASDSH---SHSDVMMTQSDPLMSQASTAVAAQN--VRRSIMLRND
PMM-SFAAQPSQGSLVNQNLLHHQHQSQNSSLSGSRALSNSVGTMGLSDSNSIGPAKHQQ
QSPANHSMQSLSDSLSGSSLYSNSLSL-PVMGHDKFPSDLDLDIFNGSLECDMESIIRSE
LMDADGLDFNFDSLISAQNVVTLN-VGNFTGAKQASSQSWVPG
>Danio_rerio
MAETTLE--PLSSLDIAIDPDFEPQKRPRSCTWPLP----ESSMAKPASNDT--DIIPEE
EDDEDD--------SAMTINANGMGEGEDNGSPSLAEE--LISINGQENTGSPLSSQAAS
ATTASPEVASQQQ--------------TPRKSSSRRNAWGNLSYADLITKAIESTPDKRL
TLSQIYDWMVSSVPYFKDKGDSNSSAGWKNSIRHNLSLHSRFIRVQNEGTGKSSWWMINP
EGGRGGKAPRRRAVSMDNGNKYTKTARGRAAKKKAALQAAQEGSLENISAGGLSKWPGSP
TSRSSDEQESCWTDFRSRTNSNASTVSGRLSPILANPELDDVS-DGPPLSPILYSSPSSL
SPPSSTSSNSKTCPGELPVLADLTGTMNLNEA--DNLMVDLLDNIATQVQ-------VVQ
ESSPFSFGSKSTASPPSTSTSQTVSNNTGGFSNPIFGPSP-GSLRQKPMQTIQENKQTSF
SSISCFET-SLQDLLNSDSLVSHSHSDVMMTQSDPLISQASAAVTSQNYRLRNNPMLRND
PMMSSFSSVLNGRSGHLQNKQHHQGSNGQGSL---RSLSNNRAQSLVNDANNLVSAKQLL
Q-PQPTMLMTETPLYSGMNG-SNGLGICHPSGTDRFPSDLDLDIFNGSLDCDVDSIIRSE
LMDSDGLDFNFDALM--QNAVSLNPVGNFTGTKQ-SNHSWVPG
```
 
- test_DBD_aligned.fasta 
  
```
>Homo_sapiens_DBD
------------------WGNLSYADLITRAIESSPDKRLTLSQIYE-------WMVRCV
PY-FKDKGDSNSSAGWKNSIRHNLSLHSRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRR
A--------------------------------
>Mus_musculus_DBD
-------------------GNLSYADLITRAIESSPDKRLTLSQIYE-------WMVRCV
PY-FKDKGDSNSSAGWKNSIRHNLSLHSRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRR
AV-------------------------------
>Danio_rerio_DBD
-------------------------------------------------------MVSSV
PY-FKDKGDSNSSAGWKNSIRHNLSLHSRFIRVQNEGTGKSSWWMINPEGGRGGKAPRRR
AVSMDNGNKYTKTARGRAAKKKAALQAAQEG--
>Gallus_gallus_DBD
-------CTVELPRLTDMAGTMNLNDGLTDNLMDDLLDNITLPSSQQSPTGGMMQRSSSF
PYGSKGSGLGSPSSSFNNAVFGPSSLNS--LR----------------------QSPMQT
I---------------------------QENKQ
>Anolis_carolinensis_DBD
SPSVNKPCNVELPRLTDMAGTMNLNDGLTDNLMDDLLDNISLPPSQQSPTGGLMQRSSSF
PYSSKGSGLGSPSSTFNNGVFGPSSLNS--LR----------------------QSPMQ-
---------------------------------
```

- test_IDR_aligned.fasta
  
```
>Homo_sapiens_IDR
MAEAPASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
EDDEDDEDGGGRAGSAMAIG--GGGGSGTLGSGLLLEDSARVLAPGGQDPGSGPATAAGG
LSGGTQALLQPQQPLPPPQPGAAGGSGQPRKCSSRRNAV-SMDNSNKYTKSRGRAAKKKA
ALQTAPESA--DDSPSQLSKWPGSPTSRSSDELDA-WTDFRSRTNSNASTVSGRLSPIMA
STELDEVQDDDAPLSPMLYSSSASLSP-----SVSKPCTVELPRLTDMAGTMNLNDGLTE
NLMDDLLDNITLP---PSQPSPTGGLMQRSSSFPYTTKGSGLGSPTSSFNSTVFGPSSLN
SLRQSPMQTIQENKPATFSSMSHYGNQTLQDLLTSDSL---SHSDVMMTQSDPLMSQAST
AVSAQN--SRRNVMLRNDPMM-SFAAQPNQGSLVNQNLLHHQHQTQ-GALGGSRALSNSV
SNMGLSESSSLGSAKHQQQSPVSQSMQTLSDSLSGSSLYSTSANLPVMGHEKFPSDLDLD
MFNGSLECDMESIIRSELMDADGLDFNFDSLISTQNVVGLN-VGNFTGAKQASSQSWVPG
>Mus_musculus_IDR
MAEAPASPVPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEE
DDDEDDEDGGGRASSAMVI---GGGVSSTLGSGLLLEDSAMLLAPGGQDLGSGPASAAGA
LSGGTPTQLQPQQPLPQPQPGAAGGSGQPRKCSSRRNAW-SMDNSNKYTKSRGRAAKKKA
ALQAAPESA--DDSPSQLSKWPGSPTSRSSDELDA-WTDFRSRTNSNASTVSGRLSPILA
STELDDVQDDDGPLSPMLYSSSASLSP-----SVSKPCTVELPRLTDMAGTMNLNDGLAE
NLMDDLLDNIALP---PSQPSPPGGLMQRGSSFPYTAKSSGLGSPTGSFNSTVFGPSSLN
SLRQSPMQTIQENRPATFSSVSHYGNQTLQDLLASDSL---SHSDVMMTQSDPLMSQAST
AVSAQN--ARRNVMLRNDPMM-SFAAQPTQGSLVNQNLLHHQHQTQ-GALGGSRALSNSV
SNMGLSDSSSLGSAKHQQQSPASQSMQTLSDSLSGSSLYSASANLPVMGHDKFPSDLDLD
MFNGSLECDMESIIRSELMDADGLDFNFDSLISTQNVVGLN-VGNFTGAKQASSQSWVPG
>Gallus_gallus_IDR
M-----------------------KNSIR------HNLSLHSRFIRVQNEGTGKS-----
--------------------------------------SWWMINPDGGKVGKAP------
----------------------------------RRRAV-SMDNSNKYTKSRGRAAKKKA
ALQTAQEAS--EDSPSQLSKWPGSPTSRSSDELDA-WTDFRSRTNSNASTISGRLSPILA
STELDDVQDDDAPLSPMLYSSPSSLSP---------------------------------
------------------------------------------------------------
----------SVNKPATFSSISHYNNQTLQDLLASDAL---SHSDVMMTQSDPLMSQAST
AVSAQN--SRRNIMLRNDPMM-SFAAQSSQGGLVNQSLPHHQHQSHSSPLSGSRALSNSI
SNIGLNDSNSLGS-KH-QQSPVNQSMQTLSDPLSGSSLYSSSMNLPVMGHEKFPSDLDLD
IFNGSLECDMESIIRSELMDADGLDFNFDSLISAQNVVSLN-VGNFTGAKQASSQSWVPG
>Anolis_carolinensis_IDR
MCIL----------------SFLSQNSIR------HNLSLHSRFIRVQNEGTGKS-----
--------------------------------------SWWVINPDGGKGGKAP------
----------------------------------RRRAV-SMDNSNKYTKSRGRAAKKKA
SLQAAQETT--EDSPTQLSKWPGSPTSRSSDELEA-WTDFRSRTNSNASTISGRLSPILA
STELDEVQDDDAPLSPMLYSSPPSMTI---------------------------------
------------------------------------------------------------
----------QENKQATFSSISHYNNQTLQDLLASDSH---SHSDVMMTQSDPLMSQAST
AVAAQN--VRRSIMLRNDPMM-SFAAQPSQGSLVNQNLLHHQHQSQNSSLSGSRALSNSV
GTMGLSDSNSIGPAKHQQQSPANHSMQSLSDSLSGSSLYSNSLSLPVMGHDKFPSDLDLD
IFNGSLECDMESIIRSELMDADGLDFNFDSLISAQNVVTLN-VGNFTGAKQASSQSWVPG
>Danio_rerio_IDR
MAET--TLEPLSSLDIAIDPDFEPQKRPRSCTWPLP----ESSMAKPASNDT--DIIPEE
EDDEDD--------SAMTINANGMGEGEDNGSPSLA--EELISINGQENTGSPLSSQAAS
ATTASPEVASQQQ--------------TPRKSSSRRNAWGNLSYADLITKAIESTPDKRL
TLSQIYDWSLENISAGGLSKWPGSPTSRSSDEQESCWTDFRSRTNSNASTVSGRLSPILA
NPELDDVS-DGPPLSPILYSSPSSLSPPSSTSSNSKTCPGELPVLADLTGTMNLNE--AD
NLMVDLLDNIATQVQVVQESSPFSFGSKSTASPPSTSTSQTVSNNTGGFSNPIFGPSP-G
SLRQKPMQTIQENKQTSFSSISCFET-SLQDLLNSDSLVSHSHSDVMMTQSDPLISQASA
AVTSQNYRLRNNPMLRNDPMMSSFSSVLNGRSGHLQNKQHHQGSNGQGSL---RSLSNNR
AQSLVNDANNLVSAKQLLQ-PQPTMLMTETPLYSGMNGSNGLGICHPSGTDRFPSDLDLD
IFNGSLDCDVDSIIRSELMDSDGLDFNFDALM--QNAVSLNPVGNFTGTKQ-SNHSWVPG
```

- test_full_final.fasta - ALIGNED AND TRIMMED
  
```
>Homo_sapiens
MAEAPASPLSPLEVELDPEFEPQSRPRSCTWPLQQASPAKPSGETASMIPEEEDDEDDSA
MAIGGGGSGTLGSGLLLEDRVLAPGGQDPGSGPATAAGGLSGGTQALLQPQQQPRKCSSR
RNAWGNLSYADLITRAIESSPDKRLTLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHN
LSLHSRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRRAVSMDNSNKYTKSRGRAAKKKAA
LQTAPESADDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTVSGRLSPIMASTEL
DEVQDDDAPLSPMLYSSSASLSPSVSKPCTVELPRLTDMAGTMNLNDGLTENLMDDLLDN
ITLPPSQPSPTGGLMQRSSSFPYTTKGSGLGSPTSSFNSTVFGPSSLNSLRQSPMQTIQE
NKPATFSSMSHYGNQTLQDLLTSDSLSHSDVMMTQSDPLMSQASTAVSAQNSRRNVMLRN
DPMMSFAAQPNQGSLVNQNLLHHQHQTQ-GALGGSRALSNSVSNMGLSESSSLGSAKHQQ
QSPVSQSMQTLSDSLSGSSLYSTSANLPVMGHEKFPSDLDLDMFNGSLECDMESIIRSEL
MDADGLDFNFDSLISTQNVVGLNVGNFTGAKQASSQSWVPG
>Mus_musculus
MAEAPASPLSPLEVELDPEFEPQSRPRSCTWPLQQASPAKPSGETASMIPEEDDDEDDSA
MVIGGGVSSTLGSGLLLEDMLLAPGGQDLGSGPASAAGALSGGTPTQLQPQQQPRKCSSR
RNAWGNLSYADLITRAIESSPDKRLTLSQIYEWMVRCVPYFKDKGDSNSSAGWKNSIRHN
LSLHSRFMRVQNEGTGKSSWWIINPDGGKSGKAPRRRAVSMDNSNKYTKSRGRAAKKKAA
LQAAPESADDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTVSGRLSPILASTEL
DDVQDDDGPLSPMLYSSSASLSPSVSKPCTVELPRLTDMAGTMNLNDGLAENLMDDLLDN
IALPPSQPSPPGGLMQRGSSFPYTAKSSGLGSPTGSFNSTVFGPSSLNSLRQSPMQTIQE
NRPATFSSVSHYGNQTLQDLLASDSLSHSDVMMTQSDPLMSQASTAVSAQNARRNVMLRN
DPMMSFAAQPTQGSLVNQNLLHHQHQTQ-GALGGSRALSNSVSNMGLSDSSSLGSAKHQQ
QSPASQSMQTLSDSLSGSSLYSASANLPVMGHDKFPSDLDLDMFNGSLECDMESIIRSEL
MDADGLDFNFDSLISTQNVVGLNVGNFTGAKQASSQSWVPG
>Gallus_gallus
M-----------------------------------------------------------
------------------------------------------------------------
-----------------------------------------------------KNSIRHN
LSLHSRFIRVQNEGTGKSSWWMINPDGGKVGKAPRRRAVSMDNSNKYTKSRGRAAKKKAA
LQTAQEASEDSPSQLSKWPGSPTSRSSDELDAWTDFRSRTNSNASTISGRLSPILASTEL
DDVQDDDAPLSPMLYSSPSSLSPSVNKPCTVELPRLTDMAGTMNLNDGLTDNLMDDLLDN
ITLPSSQQSPTGGMMQRSSSFPYGSKGSGLGSPSSSFNNAVFGPSSLNSLRQSPMQTIQE
NKQATFSSISHYNNQTLQDLLASDALSHSDVMMTQSDPLMSQASTAVSAQNSRRNIMLRN
DPMMSFAAQSSQGGLVNQSLPHHQHQSHSSPLSGSRALSNSISNIGLNDSNSLGS-KH-Q
QSPVNQSMQTLSDPLSGSSLYSSSMNLPVMGHEKFPSDLDLDIFNGSLECDMESIIRSEL
MDADGLDFNFDSLISAQNVVSLNVGNFTGAKQASSQSWVPG
>Anolis_carolinensis
M-----------------------------------------------------------
------------------------------------------------------------
------------------------------------CILSFLS----------QNSIRHN
LSLHSRFIRVQNEGTGKSSWWVINPDGGKGGKAPRRRAVSMDNSNKYTKSRGRAAKKKAS
LQAAQETTEDSPTQLSKWPGSPTSRSSDELEAWTDFRSRTNSNASTISGRLSPILASTEL
DEVQDDDAPLSPMLYSSPPSMSPSVNKPCNVELPRLTDMAGTMNLNDGLTDNLMDDLLDN
ISLPPSQQSPTGGLMQRSSSFPYSSKGSGLGSPSSTFNNGVFGPSSLNSLRQSPMQTIQE
NKQATFSSISHYNNQTLQDLLASDSHSHSDVMMTQSDPLMSQASTAVAAQNVRRSIMLRN
DPMMSFAAQPSQGSLVNQNLLHHQHQSQNSSLSGSRALSNSVGTMGLSDSNSIGPAKHQQ
QSPANHSMQSLSDSLSGSSLYSNSLSLPVMGHDKFPSDLDLDIFNGSLECDMESIIRSEL
MDADGLDFNFDSLISAQNVVTLNVGNFTGAKQASSQSWVPG
>Danio_rerio
MAETTLEPLSSLDIAIDPDFEPQKRPRSCTWPLPESSMAKPASNDTDIIPEEEDDEDDSA
MTIGMGEGEDNGSPSLAEELISINGQENTGSPLSSQAASATTASPEVASQQQTPRKSSSR
RNAWGNLSYADLITKAIESTPDKRLTLSQIYDWMVSSVPYFKDKGDSNSSAGWKNSIRHN
LSLHSRFIRVQNEGTGKSSWWMINPEGGRGGKAPRRRAVSMDNGNKYTKTRGRAAKKKAA
LQAAQEGSLESAGGLSKWPGSPTSRSSDEQESWTDFRSRTNSNASTVSGRLSPILANPEL
DDVS-DGPPLSPILYSSPSSLSPSNSKTCPGELPVLADLTGTMNLNEA--DNLMVDLLDN
IATQVQ-------VVQESSPFSFGSKSTAVSNNTGGFSNPIFGPSP-GSLRQKPMQTIQE
NKQTSFSSISCFET-SLQDLLNSDSLSHSDVMMTQSDPLISQASAAVTSQNLRNNPMLRN
DPMMSFSSVLNGRSGHLQNKQHHQGSNGQGSL---RSLSNNRAQSLVNDANNLVSAKQLL
Q-PQPTMLMTETPLYSGMNG-SNGLGIHPSGTDRFPSDLDLDIFNGSLDCDVDSIIRSEL
MDSDGLDFNFDALM--QNAVSLNVGNFTGTKQ-SNHSWVPG
```

- test_DBD_final.fasta
  
```
>Homo_sapiens_DBD
GNLSYDLITRRCVPY-FKDKGDSNSSAGWKNSISLHSRFMRKAPRRRA
>Mus_musculus_DBD
GNLSYDLITRRCVPY-FKDKGDSNSSAGWKNSISLHSRFMRKAPRRRA
>Danio_rerio_DBD
----------SSVPY-FKDKGDSNSSAGWKNSISLHSRFIRKAPRRRA
>Gallus_gallus_DBD
GTMNLDGLTDSSFPYGSKGSGLGSPSSSFNNAVSLNS--LRQSPMQTI
>Anolis_carolinensis_DBD
GTMNLDGLTDSSFPYSSKGSGLGSPSSTFNNGVSLNS--LRQSPMQ--
```

- test_IDR_final.fasta
  
```
>Homo_sapiens_IDR
MAEASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEEED
DEDDSAMAIGGGGSGTLGSGLLLSARVLAPGGQDPGSGPATAAGGLSGGTQALLQPQQQP
RKCSSRRNAVSMDNSNKYTKSRGRAAKKKAALQTAPESADDSPSQLSKWPGSPTSRSSDE
LDAWTDFRSRTNSNASTVSGRLSPIMASTELDEVQDDDAPLSPMLYSSSASLSPSVSKPC
TVELPRLTDMAGTMNLNDTENLMDDLLDNITLPPSQPSPTGGLMQRSSSFPYTTKGSGLG
SPTSSFNSTVFGPSSNSLRQSPMQTIQENKPATFSSMSHYGNQTLQDLLTSDSLSHSDVM
MTQSDPLMSQASTAVSAQNSRRNVMLRNDPMMSFAAQPNQGSLVNQNLLHHQHQTQ-GAL
GGSRALSNSVSNMGLSESSSLGSAKHQQQSPVSQSMQTLSDSLSGSSLYSTSANLPVMGH
EKFPSDLDLDMFNGSLECDMESIIRSELMDADGLDFNFDSLISTQNVVGLNVGNFTGAKQ
ASSQSWVPG
>Mus_musculus_IDR
MAEASPVPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAKPSGETAADSMIPEEDD
DEDDSAMVIGGGVSSTLGSGLLLSAMLLAPGGQDLGSGPASAAGALSGGTPTQLQPQQQP
RKCSSRRNAWSMDNSNKYTKSRGRAAKKKAALQAAPESADDSPSQLSKWPGSPTSRSSDE
LDAWTDFRSRTNSNASTVSGRLSPILASTELDDVQDDDGPLSPMLYSSSASLSPSVSKPC
TVELPRLTDMAGTMNLNDAENLMDDLLDNIALPPSQPSPPGGLMQRGSSFPYTAKSSGLG
SPTGSFNSTVFGPSSNSLRQSPMQTIQENRPATFSSVSHYGNQTLQDLLASDSLSHSDVM
MTQSDPLMSQASTAVSAQNARRNVMLRNDPMMSFAAQPTQGSLVNQNLLHHQHQTQ-GAL
GGSRALSNSVSNMGLSDSSSLGSAKHQQQSPASQSMQTLSDSLSGSSLYSASANLPVMGH
DKFPSDLDLDMFNGSLECDMESIIRSELMDADGLDFNFDSLISTQNVVGLNVGNFTGAKQ
ASSQSWVPG
>Gallus_gallus_IDR
M---------------------KNSIR------HNLSLHSRFIRVQNEGTGKS-------
-----------------------SWWMINPDGGKVGKAP---------------------
-----RRRAVSMDNSNKYTKSRGRAAKKKAALQTAQEASEDSPSQLSKWPGSPTSRSSDE
LDAWTDFRSRTNSNASTISGRLSPILASTELDDVQDDDAPLSPMLYSSPSSLSP------
------------------------------------------------------------
--------------------------SVNKPATFSSISHYNNQTLQDLLASDALSHSDVM
MTQSDPLMSQASTAVSAQNSRRNIMLRNDPMMSFAAQSSQGGLVNQSLPHHQHQSHSSPL
SGSRALSNSISNIGLNDSNSLGS-KH-QQSPVNQSMQTLSDPLSGSSLYSSSMNLPVMGH
EKFPSDLDLDIFNGSLECDMESIIRSELMDADGLDFNFDSLISAQNVVSLNVGNFTGAKQ
ASSQSWVPG
>Anolis_carolinensis_IDR
MCIL--------------SFLSQNSIR------HNLSLHSRFIRVQNEGTGKS-------
-----------------------SWWVINPDGGKGGKAP---------------------
-----RRRAVSMDNSNKYTKSRGRAAKKKASLQAAQETTEDSPTQLSKWPGSPTSRSSDE
LEAWTDFRSRTNSNASTISGRLSPILASTELDEVQDDDAPLSPMLYSSPPSMTI------
------------------------------------------------------------
--------------------------QENKQATFSSISHYNNQTLQDLLASDSHSHSDVM
MTQSDPLMSQASTAVAAQNVRRSIMLRNDPMMSFAAQPSQGSLVNQNLLHHQHQSQNSSL
SGSRALSNSVGTMGLSDSNSIGPAKHQQQSPANHSMQSLSDSLSGSSLYSNSLSLPVMGH
DKFPSDLDLDIFNGSLECDMESIIRSELMDADGLDFNFDSLISAQNVVTLNVGNFTGAKQ
ASSQSWVPG
>Danio_rerio_IDR
MAETTLEPLSSLDIAIDPDFEPQKRPRSCTWPLP----ESSMAKPASNDT--DIIPEEED
DEDDSAMTIGMGEGEDNGSPSLAEELISINGQENTGSPLSSQAASATTASPEVASQQQTP
RKSSSRRNAWNLSYADLITKAIESTPDKRLTLSQIYDWSNISAGGLSKWPGSPTSRSSDE
QESWTDFRSRTNSNASTVSGRLSPILANPELDDVS-DGPPLSPILYSSPSSLSPSNSKTC
PGELPVLADLTGTMNLNEADNLMVDLLDNIATQVQESSPFSFGSKSTASPPSTSTSQTVS
NNTGGFSNPIFGPSPGSLRQKPMQTIQENKQTSFSSISCFET-SLQDLLNSDSLSHSDVM
MTQSDPLISQASAAVTSQNLRNNPMLRNDPMMSFSSVLNGRSGHLQNKQHHQGSNGQGSL
---RSLSNNRAQSLVNDANNLVSAKQLLQ-PQPTMLMTETPLYSGMNGSNGLGICHPSGT
DRFPSDLDLDIFNGSLDCDVDSIIRSELMDSDGLDFNFDALM--QNAVSLNVGNFTGTKQ
-SNHSWVPG
```

*FULL RUN Output files:*

- full_full_final.fasta
- full_DBD_final.fasta
- full_IDR_final.fasta

  
# HMM_build.py

I utilized HMMer just to verify orthology, and if alignments are clean. I ran them for both the DBD and full sequence, keeping in mind that the disordered regions in the full sequence might affect results. I only saved the file for my DBD HMMer and posted it below. From this file I observed that EFFN was very low (2.55) for a sample size of 72, this reveals that the sequences are all very similar. This confirms (I think) that my sequences are very conserved, orthologous and likely correctly aligned. My full HMM_build file output, had an EFFN = 0.681702, for 73 sequences and was probably affected by IDR regions.

*Libraries:*

The same as previous codes.

*Test run:*

```
python3 HMM_build.py test
```

*Full run (all species):*

```
python3 HMM_build.py full
```

*Output ({name} = full or test):*

- {name}_DBD.hmm - THIS IS THE FULL RUN

```
HMMER3/f [3.4 | Aug 2023]
NAME  full_DBD_final
LENG  39
ALPH  amino
RF    no
MM    no
CONS  yes
CS    no
MAP   yes
DATE  Tue Nov 25 17:26:00 2025
NSEQ  72
EFFN  2.553223
CKSUM 3607607808
STATS LOCAL MSV       -7.5170  0.71954
STATS LOCAL VITERBI   -7.8035  0.71954
STATS LOCAL FORWARD   -4.1832  0.71954
HMM          A        C        D        E        F        G        H        I        K        L        M        N        P        Q        R        S        T        V        W        Y   
            m->m     m->i     m->d     i->m     i->i     d->m     d->d
  COMPO   2.88181  5.10076  2.56253  3.29261  3.49729  2.11057  3.72283  3.22967  2.70787  2.28252  3.60995  2.47760  2.94492  3.79174  3.21223  2.31325  3.05995  3.29634  4.52956  3.92921
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.11072  4.78091  2.33903  0.61958  0.77255  0.00000        *
      1   3.62226  4.86145  5.56221  5.04389  3.73586  5.07424  5.58312  2.06263  4.92318  1.19485  3.50666  5.21113  5.27105  5.04805  5.01412  4.47253  3.86694  0.95088  5.83365  4.72037      1 v - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      2   4.14538  5.68782  4.83339  4.82029  5.64918  4.33476  5.75686  5.49981  4.96394  4.98448  6.10808  5.02591  0.13526  5.31610  5.05189  4.35084  4.65025  5.03314  6.52916  5.79893      2 P - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      3   3.61507  5.34908  4.23530  3.65478  3.23157  4.27032  4.03946  4.02884  2.85378  3.48045  4.52342  3.93185  4.65962  3.60417  1.36096  3.68538  3.82223  3.81653  4.78246  1.05041      3 y - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      4   4.05396  5.26563  5.64122  5.20103  1.03441  5.18710  4.82625  2.82799  5.03138  1.01786  3.34905  5.16663  5.32108  4.91326  4.98171  4.60069  4.26584  3.06281  4.89808  3.37755      4 l - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      5   3.04647  5.25190  3.33778  2.85714  4.74064  3.72462  3.96131  4.13757  1.19304  3.66491  4.48884  3.06055  4.18227  3.10005  2.79601  3.06201  1.63960  3.76169  5.79492  4.51613      5 k - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      6   4.10650  6.14060  0.23543  3.21609  5.68593  4.05942  4.93889  5.51355  4.20669  4.97888  6.03774  3.79193  4.76251  4.22129  4.77929  4.02028  4.47834  5.08167  6.65032  5.51017      6 D - - -
          2.68627  4.42234  2.77528  2.73132  3.46362  2.40521  3.72503  3.29363  2.67749  2.69363  4.24086  2.90355  2.73748  3.18155  2.89809  2.37895  2.77528  2.98527  4.58486  3.61512
          0.36518  1.19917  5.40636  0.12108  2.17123  0.67570  0.71091
      7   1.48207  5.14257  3.49776  3.04344  4.79243  3.70845  4.11092  4.14469  1.09202  3.72817  4.57837  3.46410  4.23319  3.26680  2.92005  3.09825  3.34606  3.75529  5.87889  4.63058      8 k - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      8   3.94615  5.57021  4.76773  4.78686  5.78434  0.13490  5.77652  5.65341  5.06491  5.16241  6.19265  4.93571  4.97079  5.32702  5.14482  4.14985  4.48041  5.02941  6.59839  5.93552      9 G - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
      9   3.17457  5.45811  1.02461  2.68455  5.05433  3.59022  4.27504  4.49174  3.28727  4.10398  4.96797  3.22663  4.23883  3.45851  3.83295  3.15657  1.46941  4.05281  6.28703  4.86292     10 d - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.67570  0.71091
     10   2.80095  4.60989  4.00381  3.60156  4.05950  3.60726  4.49385  3.40724  3.50226  3.06599  1.57334  3.81265  4.24308  3.84580  3.77755  1.11134  3.20172  3.14002  5.56842  4.31405     11 s - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01382  4.68402  5.40636  0.61958  0.77255  0.38304  1.14503
     11   3.84606  5.71610  3.70452  3.71277  5.27138  4.11408  5.11219  5.29146  4.16801  4.82424  5.83900  0.23841  4.83019  4.46170  4.49212  3.92091  4.29352  4.80108  6.43399  5.21207     12 N - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01254  4.78091  5.50326  0.61958  0.77255  0.48576  0.95510
     12   2.98392  4.72654  4.20238  3.83215  3.95123  3.80566  4.64785  3.30190  3.72529  1.19377  4.00764  4.02206  4.42288  4.05812  3.96631  1.17601  3.37332  3.12029  5.55961  4.28457     13 s - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01254  4.78091  5.50326  0.61958  0.77255  0.48576  0.95510
     13   2.95043  5.03189  3.27300  3.14179  5.04946  3.53882  4.53760  4.63067  3.48339  4.21676  5.03223  1.21851  4.25609  3.75008  3.89879  1.08264  3.40158  4.03141  6.31401  4.96682     14 s - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01254  4.78091  5.50326  0.61958  0.77255  0.48576  0.95510
     14   4.11910  5.71821  4.95197  4.98264  5.95255  0.11152  5.94748  5.85849  5.26717  5.34401  6.38505  5.12017  5.11636  5.52296  5.32343  4.32698  4.65670  5.22048  6.72984  6.11088     15 G - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01254  4.78091  5.50326  0.61958  0.77255  0.48576  0.95510
     15   3.97180  5.25372  5.40240  4.95964  3.06106  4.93363  4.68409  3.11346  4.63515  0.96252  3.63989  4.94392  5.20851  4.77272  4.66556  4.39463  4.20055  3.26483  1.18347  3.30828     16 l - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01254  4.78091  5.50326  0.61958  0.77255  0.48576  0.95510
     16   1.97263  5.04997  3.30298  2.77960  4.56750  3.59666  3.96987  3.99630  1.35670  3.56250  4.35417  3.26305  4.07357  3.10718  3.11703  2.48089  2.19689  3.59186  5.75115  4.42061     17 k - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01254  4.78091  5.50326  0.61958  0.77255  0.48576  0.95510
     17   3.56638  6.21325  1.04645  2.55599  5.53330  3.67927  4.34920  5.16587  3.50637  4.61932  5.52151  1.13114  4.36907  3.53759  4.18941  3.41192  3.89452  4.67953  6.72041  5.13777     18 d - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01281  4.75996  5.48231  0.61958  0.77255  0.48576  0.95510
     18   2.93659  5.02066  3.25113  3.11662  5.02514  3.52519  4.51169  4.60409  3.45410  4.19010  5.00599  1.20624  4.23920  3.72284  3.87069  1.11752  3.38484  4.01054  6.28919  4.94073     19 s - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01281  4.75996  5.48231  0.61958  0.77255  0.48576  0.95510
     19   3.93864  5.13770  5.87462  5.35225  3.60273  5.40334  5.81488  1.05279  5.21389  0.91782  3.35146  5.55823  5.47254  5.19452  5.22560  4.83194  4.16463  2.40686  5.84287  4.82046     20 l - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01281  4.75996  5.48231  0.61958  0.77255  0.48576  0.95510
     20   3.39840  5.17904  4.25986  3.56433  4.24646  4.15091  4.24427  3.60185  2.64166  3.03616  1.36433  3.88694  4.54557  3.46507  1.18061  3.55026  3.61654  3.46689  5.61115  4.41857     21 r - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01281  4.75996  5.48231  0.61958  0.77255  0.48576  0.95510
     21   3.40146  6.04731  1.28231  2.08771  5.29390  3.65475  1.32663  4.86752  3.21100  4.32811  5.17284  3.10124  4.27878  3.35478  3.81666  3.26818  3.68722  4.41793  6.45653  4.92027     22 d - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01281  4.75996  5.48231  0.61958  0.77255  0.48576  0.95510
     22   3.56638  6.21325  1.04645  2.55599  5.53330  3.67927  4.34920  5.16587  3.50637  4.61932  5.52151  1.13114  4.36907  3.53759  4.18941  3.41192  3.89452  4.67953  6.72041  5.13777     23 d - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.03359  4.75996  3.71066  0.61958  0.77255  0.48576  0.95510
     23   4.19189  5.38754  5.88503  5.39399  2.67482  5.45195  5.44914  2.72353  5.23898  0.39075  3.21642  5.55365  5.47777  5.08622  5.18096  4.90478  4.39663  3.03010  5.42243  4.16246     24 L - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01307  4.73945  5.46180  0.61958  0.77255  0.52981  0.88847
     24   2.96857  4.70876  4.17816  3.80390  3.92622  3.79349  4.62104  3.27379  3.69675  1.18721  3.98076  3.99963  4.40579  4.03016  3.93956  1.21270  3.35438  3.09353  5.53366  4.25914     25 l - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01284  4.75710  5.47944  0.61958  0.77255  0.52981  0.88847
     25   3.38536  5.61036  3.01963  2.87996  4.37435  3.80975  1.26350  4.57621  3.06822  4.05276  4.95535  1.23579  4.39168  3.47606  3.44878  3.35228  3.68288  4.18913  5.75336  4.21960     26 n - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01284  4.75710  5.47944  0.61958  0.77255  0.52981  0.88847
     26   2.95131  4.62995  4.35050  3.97197  4.05500  3.83402  4.75934  1.21412  3.88562  2.98285  4.04815  4.11122  4.45409  4.18842  4.11311  1.22657  3.33985  2.67251  5.64976  4.40483     27 i - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01277  4.76255  5.48490  0.61958  0.77255  0.52981  0.88847
     27   2.87333  5.12894  3.26193  2.72769  4.55718  3.63498  3.90641  3.98799  2.58849  3.53516  4.32911  2.82133  3.34200  3.03824  1.48278  2.69722  1.82872  3.60287  5.71135  4.38151     28 r - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01277  4.76255  5.48490  0.61958  0.77255  0.52981  0.88847
     28   4.15793  5.35362  5.78883  5.33997  1.13096  5.32314  4.98481  2.85828  5.17373  0.87579  3.35706  5.32137  5.42056  5.01974  5.10364  4.74395  4.36473  3.12381  5.03042  3.54080     29 l - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01277  4.76255  5.48490  0.61958  0.77255  0.52981  0.88847
     29   2.91317  4.42916  4.23112  3.65857  3.50557  3.99176  4.31185  2.17837  3.36130  2.44581  1.89825  3.92543  1.69369  3.79886  3.75967  3.28174  3.15195  2.40195  5.06768  3.87628     30 p - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.04435  4.76255  3.35699  0.61958  0.77255  0.52981  0.88847
     30   2.79350  4.90322  3.31290  3.00238  4.81653  2.12487  4.29220  4.27616  3.16860  3.86505  4.65685  1.28805  4.11872  3.46675  3.60806  2.61880  1.83336  3.75504  6.05509  4.73863     31 n - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01318  4.73138  5.45373  0.61958  0.77255  0.59307  0.80436
     31   2.19631  4.64484  3.90265  3.48706  4.78474  2.94440  4.56678  4.22766  3.50021  3.86896  4.66222  3.69314  1.25323  3.79211  3.86414  1.73329  2.16007  3.65544  6.08760  4.85773     32 p - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01318  4.73138  5.45373  0.61958  0.77255  0.59307  0.80436
     32   3.12140  5.69225  1.44539  2.09532  5.01389  3.60908  3.98711  4.51992  1.44369  3.98448  4.76898  3.06312  4.13519  3.11653  3.36192  2.76504  3.37672  4.07602  6.11640  4.65633     33 k - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01318  4.73138  5.45373  0.61958  0.77255  0.59307  0.80436
     33   2.80955  4.74309  4.01362  3.98642  5.27429  0.43842  5.14161  4.84662  4.29064  4.51051  5.31501  4.00508  4.28971  4.48450  4.51038  2.41769  3.36982  4.05334  6.56241  5.42069     34 G - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.04137  4.73138  3.45097  0.61958  0.77255  0.59307  0.80436
     34   2.70276  4.65213  4.00309  3.88622  5.17800  1.00862  5.00571  4.70463  4.10340  4.35996  5.14405  3.91014  4.19991  4.31219  4.36089  1.08652  3.25119  3.92989  6.47685  5.31752     35 g - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01356  4.70356  5.42591  0.61958  0.77255  0.64483  0.74392
     35   3.22775  5.41950  3.44470  2.98604  4.98178  1.48595  4.03451  4.38315  1.14997  3.84786  4.68473  3.43738  4.29785  3.17216  2.60001  3.22986  3.47044  3.99612  5.91180  4.67831     36 k - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01356  4.70356  5.42591  0.61958  0.77255  0.64483  0.74392
     36   2.73822  4.52395  3.87033  3.39481  3.94191  2.38635  4.30522  3.28591  3.34612  1.43810  3.91592  3.67456  4.16805  3.64822  3.66019  1.60456  3.08874  2.88079  5.39405  4.16757     37 l - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01356  4.70356  5.42591  0.61958  0.77255  0.55305  0.85612
     37   4.03054  5.64254  4.85792  4.88286  5.86695  0.12286  5.86054  5.75411  5.16430  5.25173  6.28718  5.02599  5.04203  5.42324  5.23265  4.23628  4.56653  5.12302  6.66295  6.02164     38 G - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01318  4.73138  5.45373  0.61958  0.77255  0.59307  0.80436
     38   3.09158  5.25988  3.41106  2.99185  4.90973  3.72929  4.10429  4.34880  1.17669  3.85664  4.68355  3.43477  4.25114  3.25088  2.91979  1.33918  3.38998  3.92229  5.94620  4.67413     39 k - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.01318  4.73138  5.45373  0.61958  0.77255  0.59307  0.80436
     39   0.95707  4.67041  4.19298  3.99633  5.00919  3.44978  5.00123  4.39752  4.06056  4.12174  4.97037  3.99289  1.17094  4.32189  4.29915  2.94839  3.28327  3.78606  6.34507  5.19903     40 a - - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.00889  4.72709        *  0.61958  0.77255  0.00000        *
//
```

- {name}_DBD_hmmsearch.tbl

```
#                                                                     --- full sequence ---- --- best 1 domain ---- --- domain number estimation ----
# target name              accession  query name           accession    E-value  score  bias   E-value  score  bias   exp reg clu  ov env dom rep inc description of target
#      ------------------- ---------- -------------------- ---------- --------- ------ ----- --------- ------ -----   --- --- --- --- --- --- --- --- ---------------------
Elephas_maximus            -          full_DBD_final       -            9.2e-30   94.7  17.3   6.4e-17   53.6   0.3   5.4   5   0   0   5   5   4   2 -
Loxodonta_africana         -          full_DBD_final       -            9.2e-30   94.7  17.3   6.4e-17   53.6   0.3   5.4   5   0   0   5   5   4   2 -
Rhinolophus_ferrumequinum  -          full_DBD_final       -            2.3e-29   93.4  19.7   6.4e-17   53.6   0.3   5.3   5   0   0   5   5   4   2 -
Elephantulus_edwardii      -          full_DBD_final       -            2.4e-29   93.3  22.8   6.4e-17   53.6   0.3   5.6   6   0   0   6   6   4   2 -
Dasypus_novemcinctus       -          full_DBD_final       -            3.4e-29   92.9  19.3   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Phoca_vitulina             -          full_DBD_final       -            3.7e-29   92.8  20.8   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Monachus_schauinslandi     -          full_DBD_final       -            3.7e-29   92.8  20.8   6.4e-17   53.6   0.3   5.4   5   0   0   5   5   4   2 -
Balaenoptera_acutorostrata -          full_DBD_final       -              4e-29   92.6  18.4   6.4e-17   53.6   0.3   5.4   5   0   0   5   5   4   2 -
Homo_sapiens               -          full_DBD_final       -              6e-29   92.1  18.5   6.4e-17   53.6   0.3   5.6   5   0   0   5   5   4   2 -
Pan_troglodytes            -          full_DBD_final       -            6.1e-29   92.1  18.5   6.4e-17   53.6   0.3   5.6   5   0   0   5   5   4   2 -
Nomascus_leucogenys        -          full_DBD_final       -            6.5e-29   92.0  18.5   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Chlorocebus_sabaeus        -          full_DBD_final       -            6.6e-29   92.0  18.5   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Macaca_mulatta             -          full_DBD_final       -            6.6e-29   92.0  18.5   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Papio_anubis               -          full_DBD_final       -            6.6e-29   92.0  18.5   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Anas_platyrhynchos         -          full_DBD_final       -            7.2e-29   91.8  18.9   1.9e-16   52.1   0.2   5.4   5   0   0   5   5   4   2 -
Macropus_eugenii           -          full_DBD_final       -            8.6e-29   91.6  19.2   4.4e-16   50.9   0.3   5.6   5   0   0   5   5   4   3 -
Naja_naja                  -          full_DBD_final       -            1.5e-28   90.8  20.6   1.9e-16   52.1   0.2   5.3   5   0   0   5   5   4   2 -
Gorilla_gorilla            -          full_DBD_final       -            2.7e-28   90.0  18.1   6.4e-17   53.6   0.3   5.7   5   0   0   5   5   4   2 -
Callithrix_jacchus         -          full_DBD_final       -            4.5e-28   89.3  18.9   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Heterocephalus_glaber      -          full_DBD_final       -            6.7e-28   88.7  23.1   6.4e-17   53.6   0.3   5.6   6   0   0   6   6   4   2 -
Strigops_habroptilus       -          full_DBD_final       -            7.1e-28   88.6  20.5   2.6e-16   51.7   0.2   5.2   5   0   0   5   5   4   2 -
Melopsittacus_undulatus    -          full_DBD_final       -            7.4e-28   88.6  20.5   2.6e-16   51.7   0.2   5.1   5   0   0   5   5   4   2 -
Ictidomys_tridecemlineatus -          full_DBD_final       -            8.8e-28   88.4  25.7   6.4e-17   53.6   0.3   5.7   6   0   0   6   6   4   2 -
Marmota_marmota            -          full_DBD_final       -            8.8e-28   88.4  25.7   6.4e-17   53.6   0.3   5.7   6   0   0   6   6   4   2 -
Physeter_macrocephalus     -          full_DBD_final       -            9.7e-28   88.2  23.0   6.4e-17   53.6   0.3   5.5   6   0   0   6   6   4   2 -
Tursiops_truncatus         -          full_DBD_final       -              1e-27   88.2  23.7   5.7e-17   53.8   0.3   5.5   6   0   0   6   6   4   3 -
Canis_lupus_familiaris     -          full_DBD_final       -            1.1e-27   88.1  25.9   6.4e-17   53.6   0.3   5.7   6   0   0   6   6   4   2 -
Saimiri_boliviensis        -          full_DBD_final       -            1.1e-27   88.0  20.1   6.4e-17   53.6   0.3   5.5   5   0   0   5   5   4   2 -
Bos_taurus                 -          full_DBD_final       -            1.3e-27   87.8  23.7   6.4e-17   53.6   0.3   5.7   6   0   0   6   6   4   2 -
Ovis_aries                 -          full_DBD_final       -            1.3e-27   87.8  23.7   6.4e-17   53.6   0.3   5.7   6   0   0   6   6   4   2 -
Balaenoptera_musculus      -          full_DBD_final       -            1.5e-27   87.6  23.7   6.4e-17   53.6   0.3   5.6   6   0   0   6   6   4   2 -
Capra_hircus               -          full_DBD_final       -            1.5e-27   87.6  24.1   6.4e-17   53.6   0.3   5.8   6   0   0   6   6   4   2 -
Eubalaena_glacialis        -          full_DBD_final       -            1.6e-27   87.6  23.7   6.4e-17   53.6   0.3   5.5   6   0   0   6   6   4   2 -
Orcinus_orca               -          full_DBD_final       -            1.6e-27   87.5  23.7   6.4e-17   53.6   0.3   5.5   6   0   0   6   6   4   2 -
Microcebus_murinus         -          full_DBD_final       -            1.8e-27   87.3  25.1   6.4e-17   53.6   0.3   5.7   6   0   0   6   6   4   2 -
Ambystoma_mexicanum        -          full_DBD_final       -            2.7e-27   86.8   9.1   5.1e-16   50.7   0.3   3.9   3   0   0   3   3   3   2 -
Sus_scrofa                 -          full_DBD_final       -            3.2e-27   86.6  25.2   6.4e-17   53.6   0.3   5.6   6   0   0   6   6   4   2 -
Otolemur_garnettii         -          full_DBD_final       -            6.2e-27   85.7  25.4   6.4e-17   53.6   0.3   5.6   6   0   0   6   6   4   2 -
Cavia_porcellus            -          full_DBD_final       -            6.7e-27   85.5  26.1   6.3e-17   53.6   0.3   5.7   7   0   0   7   7   4   2 -
Myotis_brandtii            -          full_DBD_final       -            1.4e-26   84.5  25.9   2.4e-16   51.8   0.5   5.7   6   0   0   6   6   4   3 -
Rattus_norvegicus          -          full_DBD_final       -            2.2e-26   83.9  21.7   6.4e-17   53.6   0.3   5.4   6   0   0   6   6   4   2 -
Mus_musculus               -          full_DBD_final       -            4.3e-26   83.0  21.7   6.4e-17   53.6   0.3   5.3   6   0   0   6   6   4   2 -
Erinaceus_europaeus        -          full_DBD_final       -            6.2e-26   82.4  19.9   6.5e-17   53.6   0.3   5.3   6   0   0   6   6   4   2 -
Myotis_myotis              -          full_DBD_final       -            7.9e-26   82.1  25.9   3.6e-16   51.2   0.5   5.6   6   0   0   6   6   4   2 -
Myotis_lucifugus           -          full_DBD_final       -            7.9e-26   82.1  25.9   3.6e-16   51.2   0.5   5.6   6   0   0   6   6   4   2 -
Taeniopygia_guttata        -          full_DBD_final       -            9.4e-26   81.9  27.9   2.6e-16   51.6   0.2   5.5   6   0   0   6   6   4   2 -
Lepisosteus_oculatus       -          full_DBD_final       -            1.4e-25   81.3  18.7     2e-16   52.0   0.2   5.0   5   0   0   5   5   4   2 -
Oncorhynchus_mykiss        -          full_DBD_final       -            1.8e-25   80.9  13.2   1.9e-16   52.1   0.2   5.0   5   0   0   5   5   4   2 -
Xenopus_tropicalis         -          full_DBD_final       -            4.3e-25   79.7  16.0   6.7e-17   53.5   0.3   5.0   4   0   0   4   4   4   2 -
Monodelphis_domestica      -          full_DBD_final       -            9.6e-25   78.6  29.3   6.3e-16   50.4   0.3   5.7   6   0   0   6   6   4   3 -
Oryzias_latipes            -          full_DBD_final       -            7.7e-24   75.8  16.1   4.3e-16   51.0   0.2   4.7   4   0   0   4   4   3   2 -
Latimeria_chalumnae        -          full_DBD_final       -            9.9e-24   75.4  20.5   5.3e-16   50.7   0.3   4.8   5   0   0   5   5   3   2 -
Xenopus_laevis             -          full_DBD_final       -            2.7e-23   74.0  15.0   6.7e-17   53.5   0.3   4.9   4   0   0   4   4   4   2 -
Rana_catesbeiana           -          full_DBD_final       -            1.3e-22   71.8  16.5   1.8e-16   52.2   0.2   4.5   5   0   0   5   5   3   2 -
Fukomys_damarensis         -          full_DBD_final       -            1.1e-21   68.8  27.9     5e-12   38.0   2.0   6.5   7   0   0   7   7   5   3 -
Danio_rerio                -          full_DBD_final       -            1.3e-21   68.6   5.0   1.9e-16   52.1   0.2   4.1   3   0   0   3   3   3   2 -
Gallus_gallus              -          full_DBD_final       -              1e-17   56.2  18.1   1.3e-12   39.8   2.0   5.4   5   0   0   5   5   4   2 -
Columba_livia              -          full_DBD_final       -            1.6e-16   52.3  19.3   3.4e-12   38.5   1.7   5.3   5   0   0   5   5   4   2 -
Equus_caballus             -          full_DBD_final       -            1.8e-16   52.1  23.3   4.6e-12   38.1   1.4   5.7   6   0   0   6   6   4   3 -
Tupaia_belangeri           -          full_DBD_final       -            3.2e-16   51.4  18.0   1.7e-12   39.5   2.2   4.8   5   0   0   5   5   3   3 -
Fulmarus_glacialis         -          full_DBD_final       -            5.4e-16   50.6  19.3   2.8e-12   38.7   1.7   5.4   5   0   0   5   5   4   2 -
Python_bivittatus          -          full_DBD_final       -            8.6e-16   50.0  22.8   2.9e-12   38.7   1.7   5.4   5   0   0   5   5   4   2 -
Crocodylus_porosus         -          full_DBD_final       -            3.4e-15   48.1  26.8   8.1e-13   40.5   2.5   5.6   6   0   0   6   6   5   2 -
Sorex_araneus              -          full_DBD_final       -            6.8e-15   47.1  27.5   2.8e-12   38.8   2.3   6.0   7   0   0   7   7   4   3 -
Callorhinchus_milii        -          full_DBD_final       -            1.2e-14   46.4  16.0   3.6e-11   35.2   1.9   5.1   4   0   0   4   4   4   2 -
Anolis_carolinensis        -          full_DBD_final       -            1.9e-14   45.7  23.2   1.1e-11   36.8   2.0   5.4   5   0   0   5   5   4   2 -
Ursus_arctos               -          full_DBD_final       -            3.3e-14   44.9  15.4   4.4e-12   38.1   3.0   4.6   4   0   0   4   4   3   2 -
Vulpes_vulpes              -          full_DBD_final       -            3.3e-14   44.9  15.4   4.4e-12   38.1   3.0   4.6   4   0   0   4   4   3   2 -
Felis_catus                -          full_DBD_final       -            4.8e-14   44.4  15.4   4.4e-12   38.1   3.0   4.5   4   0   0   4   4   3   2 -
Tarsius_syrichta           -          full_DBD_final       -            1.1e-12   40.1  18.6   3.5e-12   38.4   1.2   4.8   5   0   0   5   5   3   2 -
Corvus_brachyrhynchos      -          full_DBD_final       -            2.6e-12   38.8   1.7   2.6e-12   38.8   1.7   4.8   5   0   0   5   5   3   1 -
Eptesicus_fuscus           -          full_DBD_final       -            7.6e-12   37.4  19.7   2.6e-11   35.6   1.8   4.7   5   0   0   5   5   3   2 -
Chrysemys_picta            -          full_DBD_final       -            2.3e-08   26.3   0.1   5.6e-08   25.0   0.1   1.7   1   0   0   1   1   1   1 -
#
# Program:         hmmsearch
# Version:         3.4 (Aug 2023)
# Pipeline mode:   SEARCH
# Query file:      full_DBD.hmm
# Target file:     full_foxo3_seq.fasta
# Option settings: hmmsearch --tblout full_DBD_hmmsearch.tbl full_DBD.hmm full_foxo3_seq.fasta 
# Current dir:     /mnt/c/Users/Erika/Desktop/BMEProject/BME-Term-Project
# Date:            Tue Nov 25 17:26:00 2025
# [ok]
```

# Phyl_tree.py

I replicated the code created by Osbourne and others in python because I'm more comfortable with it. To create my phylogenetic tree I use a distance based method called neighbor joining. I felt this was a good choice considering again, my smaller 73 sequence sample size.


*Libraries:*
- AlignIO
- DistanceCalculator
- DistanceTreeConstructor
- Phylo
- matplotlib.pyplot

*RUN:*
```
python3 Phyl_tree
```

# basic download information #
Biopython:

```
pip install biopython
```

Requests:

```
pip install requests
```

MAFFT:

```
sudo apt-get install mafft
```

Trimal:
```
git clone https://github.com/inab/trimal.git

cd trimal
cd Source

make
```
HMMer:
```
sudo apt install hmmer
```

Phylogenic tree
```
pip install biopython scikit-bio
pip install matplotlib
```

**other sorting stuff this part is not organized at all**
```
sudo apt install -y python3 python3-pip

sudo apt install python3.12-venv

python3 -m venv venv

source venv/bin/activate

sudo apt update
sudo apt install -y build-essential git

git clone https://github.com/inab/trimal.git
cd trimal
cd Source

# compile
make

# move binary to PATH
sudo cp trimal /usr/local/bin/
sudo chmod +x /usr/local/bin/trimal

# test
trimal -h

pip install biopython
```
**CITATIONS**

Kazutaka K, Daron M. S. 2013. MAFFT Multiple Sequence Alignment Software Version 7: Improvements in Performance and Usability, Molecular Biology and Evolution. 30:4(772–780). https://doi.org/10.1093/molbev/mst010

Capella-Gutiérrez S, Silla-Martínez M, Gabaldón T. 2009. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics. 25:15(1972–1973). https://doi.org/10.1093/bioinformatics/btp348
